ios/FluentUI/Vnext/HUD/HeadsUpDisplay.swift

```diff
@@ -0,0 +1,227 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import SwiftUI
+import UIKit
+
+/// Defines the content type of the Heads-up display.
+/// The `.activity` value will make the HUD display an Activity Indicator.
+/// The `.success` and `.failure` values will make the HUD display its default images for the cases.
+/// The `.custom` value allows a UIImage to be passed as a parameter for the HUD to display.
+public enum HUDType: Equatable, Hashable {
+    case activity
+    case custom(image: UIImage)
+    case failure
+    case success
+
+    public func hash(into hasher: inout Hasher) {
+        switch self {
+        case .custom(let image):
+            hasher.combine(image)
+        case .activity, .success, .failure:
+                break
+        }
+    }
+}
+
+/// Properties available to customize the state of the Heads-up Display state.
+public protocol MSFHUDState {
+
+    /// Label string presented by the Heads-up Display.
+    var label: String? { get set }
+
+    /// The action executed when the Heads up display is tapped.
```

Comment:
Comments are pretty inconsistent about spelling, hyphenation, and capitalization:
* Heads-up Display
* Heads up display
* Heads-up display
* HeadsUpDisplay

Which should be the official way to describe it?

Reply:
Sounds good. I'll update all of them to "Heads-up display".